x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicy.java

```diff
@@ -0,0 +1,236 @@
+            in.readString(),
+            in.readStringList(),
+            in.readTimeValue()
+        );
+    }
+
+    public EnrichPolicy(Type type,
+                        QuerySource query,
+                        String indexPattern,
+                        String enrichKey,
+                        List<String> enrichValues,
+                        TimeValue schedule) {
+        this.type = type;
+        this.query= query;
+        this.schedule = schedule;
+        this.indexPattern = indexPattern;
+        this.enrichKey = enrichKey;
+        this.enrichValues = enrichValues;
+    }
+
+    public Type getType() {
+        return type;
+    }
+
+    public QuerySource getQuery() {
+        return query;
+    }
+
+    public String getIndexPattern() {
+        return indexPattern;
+    }
+
+    public String getEnrichKey() {
+        return enrichKey;
+    }
+
+    public List<String> getEnrichValues() {
+        return enrichValues;
+    }
+
+    public TimeValue getSchedule() {
+        return schedule;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(type.toString());
+        out.writeBytesReference(query.query);
+        out.writeEnum(query.contentType);
+        out.writeString(indexPattern);
+        out.writeString(enrichKey);
+        out.writeStringCollection(enrichValues);
+        out.writeTimeValue(schedule);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field(TYPE.getPreferredName(), type.toString());
+        builder.field(QUERY.getPreferredName(), query.getQueryAsMap());
+        builder.field(INDEX_PATTERN.getPreferredName(), indexPattern);
+        builder.field(ENRICH_KEY.getPreferredName(), enrichKey);
+        builder.array(ENRICH_VALUES.getPreferredName(), enrichValues.toArray(new String[0]));
+        builder.field(SCHEDULE.getPreferredName(), schedule.getStringRep());
+        return builder;
+    }
+
+    @Override
+    public boolean isFragment() {
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        EnrichPolicy policy = (EnrichPolicy) o;
+        return type == policy.type &&
+            query.equals(policy.query) &&
+            indexPattern.equals(policy.indexPattern) &&
+            enrichKey.equals(policy.enrichKey) &&
+            enrichValues.equals(policy.enrichValues) &&
+            schedule.equals(policy.schedule);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+            type,
+            query,
+            indexPattern,
+            enrichKey,
+            enrichValues,
+            schedule
+        );
+    }
+
+    public String toString() {
+        return Strings.toString(this);
+    }
+
+    public enum Type {
+
+        EXACT_MATCH;
```

Comment:
Using an enum here can be problematic for future mixed clusters. For example say we add "geo" the old node would blow up with IllegalArgument exception, but if it were it were a String it would be fine (even if it didn't know what to do with it). 

I think this was a similar problem when the docker build type was introduced.

Reply:
Yes, this is problematic, because enrich policies are stored in the cluster state. I will change it into a string.